test/gateway/ost_prime/initialize.js

```diff
@@ -0,0 +1,96 @@
+const OSTPrime = artifacts.require("OSTPrime")
+  , BN = require('bn.js');
+
+const Utils = require('../../../test/test_lib/utils');
+
+contract('OSTPrime.sol', function (accounts) {
+
+  describe('initialize', async () => {
+
+    const DECIMAL = new BN(10);
+    const POW = new BN(18);
+    const DECIMAL_FACTOR = DECIMAL.pow(POW);
+    const TOKENS_MAX = new BN(800000000).mul(DECIMAL_FACTOR);
+
+    let brandedTokenAddress, ostPrime;
+
+    beforeEach(async function () {
+      brandedTokenAddress = accounts[2];
+      ostPrime = await OSTPrime.new(brandedTokenAddress);
+    });
+
+    it('The balance of OST prime contract must be zero', async function () {
+
+      let balance = await Utils.getBalance(ostPrime.address);
+      assert.strictEqual(
+        new BN(0).eq(new BN(balance)),
+        true,
```

Comment:
1. You don't need to convert to `BN` if none of the values to compare were `BN` before.
2. *If* you use `BN`, you can `assert(one.eq(two), 'message')`

It seems this should simply be:
```js
      let balance = await Utils.getBalance(ostPrime.address);
      assert.strictEqual(
        0,
        balance,
        `The balance of the contract must be zero after initialization.`
      );
```

Generally, I am not sure this test is sensible 🤔

Reply:
Initialize method requires that the payable amount must be equal to the TOKEN_MAX. 
So shouldn't it be tested that the initial amount before the initialization should be zero? Else after initialization, the contract may contain more than TOKEN_MAX.
Let me know if this sounds sensible, If not I can remove this.

About converting to BN:

That will not work. `getBalance` returns `string`. 
Changed to 
```
assert.strictEqual(
      balance,
      '0',
      `The balance of contract must be zero.`,
    );
```